src/common/ceph_context.cc

```diff
@@ -569,6 +578,17 @@ PerfCountersCollection *CephContext::get_perfcounters_collection()
   return _perf_counters_collection;
 }
 
+void CephContext::enable_perf_counter()
+{
+  ceph_spin_lock(&_cct_perf_lock);
```

Comment:
Yeah... the purpose was to protect _cct_perf with the service thread (which check if it is NULL) and other threads which calls 'enable_perf_counter' (which assign this variable).

We should be able to remove it with a comment of this method to let the caller call this before starting the service thread and there would no contention.

Let me remove it and add a clear function comment.